pkg/image/registry/image/strategy.go

```diff
@@ -57,7 +57,9 @@ func (imageStrategy) AllowUnconditionalUpdate() bool {
 }
 
 // PrepareForUpdate clears fields that are not allowed to be set by end users on update.
-// It extracts the latest info from the manifest and sets that on the object.
+// It extracts the latest info from the manifest and sets that on the object. It allows a user
+// to update the manifest so that it matches the digest (in case an older server stored a manifest
+// that was malformed, it can always be corrected).
```

Comment:
There already is one, pkg/image/registry/image/etcd#TestUpdateResetsMetadata and TestCreateSetsMetadata